test/integration/rails_integration_test.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+require "test_helper"
+require "tempfile"
+require "fileutils"
+require "open3"
+
+class RailsIntegrationTest < ActiveSupport::TestCase
+  CommandFailed = Class.new(StandardError)
+
+  test "running a job in a Rails app works" do
+    # This test is slow, because it has to install gems and generate an entire Rails app.
+    in_dummy_rails_app_with_job_iteration do |_dir|
+      puts "Writing app/jobs/example_job.rb" if ENV["DEBUG"]
+      File.write("app/jobs/example_job.rb", <<~RUBY)
+        class ExampleJob < ApplicationJob
+          include JobIteration::Iteration
+
+          def build_enumerator(cursor:)
+            enumerator_builder.array(["it works"], cursor: cursor)
+          end
+
+          def each_iteration(string)
+            puts string
+          end
+        end
+      RUBY
+
+      output = run_or_raise("bin/rails", "runner", "ExampleJob.perform_now")
+      assert_includes(output, "it works")
+    end
+  end
+
+  private
+
+  def in_dummy_rails_app_with_job_iteration
+    Dir.mktmpdir do |dir|
+      Dir.chdir(dir) do |dir|
+        dir = File.join(dir, "dummy")
+        FileUtils.mkdir(dir)
+        Dir.chdir(dir) do |dir|
+          with_env(ENV.reject { |k, _v| k == "BUNDLE_GEMFILE" }) do # Don't use the Gemfile override set in CI.
+            run_or_raise("bundle", "init")
+
+            # It is unclear why we need to do this, but otherwise it blows up with:
+            #   Could not find <the gems below> in any of the sources (Bundler::GemNotFound)
+            run_or_raise("gem", "install", "nio4r")
+            run_or_raise("gem", "install", "websocket-driver")
+            run_or_raise("gem", "install", "date")
+            run_or_raise("gem", "install", "racc")
+
+            # It is also unclear why we need to --skip-install, lock, and install, instead of just using `bundle add`,
+            # but, again, otherwise it blows up with the same error as above.
+            run_or_raise("bundle", "add", "rails", "--version", rails_version.to_s, "--skip-install")
+            run_or_raise("bundle", "lock")
+            run_or_raise("bundle", "install")
```

Comment:
There's something really weird going on here. I had to add all of this for it to run locally, but we still run into this error in CI, just for different gems.

I think there's something weird going on, to do with Bundler/Rubygems environment variables or something, and the interaction with running `bundler` within this app.